tests/validate/bandwidth_test/src/host.cpp

```diff
@@ -66,26 +66,36 @@ int main(int argc, char** argv) {
         }
     }
 
-    int NUM_KERNEL;
+    int num_kernel = 0, num_kernel_ddr = 0;
+    bool chk_hbm_mem = false;
     std::string filename = "/platform.json";
     std::string platform_json = test_path + filename;
 
     try {
-        boost::property_tree::ptree loadPtreeRoot;
-        boost::property_tree::read_json(platform_json, loadPtreeRoot);
-        boost::property_tree::ptree temp;
-
-        temp = loadPtreeRoot.get_child("total_ddr_banks");
-        NUM_KERNEL = temp.get_value<int>();
+        boost::property_tree::ptree load_ptree_root;
+        boost::property_tree::read_json(platform_json, load_ptree_root);
+
+        auto temp = load_ptree_root.get_child("total_ddr_banks");
+        num_kernel = temp.get_value<int>();
+        num_kernel_ddr = num_kernel;
+        auto pt_mem_array = load_ptree_root.get_child("meminfo");
+        for (const auto& mem_entry : pt_mem_array) {
+            boost::property_tree::ptree pt_mem_entry = mem_entry.second;
+            auto sValue = pt_mem_entry.get<std::string>("type");
+            if (sValue == "HBM") {
+                chk_hbm_mem = true;
+            }
+        }
+        if (chk_hbm_mem) {
+            num_kernel_ddr = num_kernel - 1;
+        }
     } catch (const std::exception& e) {
         std::string msg("ERROR: Bad JSON format detected while marshaling build metadata (");
         msg += e.what();
         msg += ").";
         std::cout << msg << std::endl;
     }
```

Comment:
Hmm...If an error was found, shouldn't the code stop testing this test by throwing an exception?